File: back-end/taskManager/Domain/Task.cs
using task_manager.Domain.Enums;

namespace task_manager.Domain;

public class Task
{
    public Guid TaskId { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public EnumTaskStatus? Status { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? CompletedDate { get; set; }
    public Guid UserId { get; set; }
    public User? User { get; set; }
    public Guid ProjectId { get; set; }
    public Project? Project { get; set; }

}

File: front-end/src/app/shared/Models/Task.ts
export interface Task{
 taskId?: string;
 userId?: string;
 projectId?: string;
 title?: string;
 description?: string;
 status?: EnumTaskStatus;
 dueDate?: string;
 startDate?: string;
 completedDate?: string;
}

export enum EnumTaskStatus {
  NotStarted = 0,
  Postponed = 1,
  Progress = 2,
  Done = 3
}
